Build Fails on Mac OS X

Currently compilation is failing on Mac OS X using clang. The important bits of the error are below:

```
In file included from src/ripple_data/crypto/StellarPrivateKey.cpp:1:
In file included from src/ripple_data/crypto/StellarPrivateKey.h:4:
In file included from src/ripple_data/crypto/../protocol/RippleAddress.h:23:
In file included from src/ripple_data/crypto/../protocol/../crypto/Base58Data.h:35:
src/ripple_data/crypto/../../ripple/types/api/Base58.h:86:24: error: implicit instantiation of undefined template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    static std::string encode (InputIt first, InputIt last,
                       ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/c++/v1/iosfwd:188:33: note: template is declared here
    class _LIBCPP_TYPE_VIS_ONLY basic_string;
                                ^
```

I assume building on linux with gcc/g++ works, and a quick google search leads me to believe this is one of the nuances of clang. I'm working on a fix and will submit a PR soon.

--- a/src/ripple/types/api/Base58.h
+++ b/src/ripple/types/api/Base58.h
@@ -33,6 +33,7 @@
 #define RIPPLE_TYPES_BASE58_H
 
 #include <iterator>
+#include <string>
 #include <type_traits>
 
 #include "Blob.h"

